src/ssvc/selection.py

```diff
@@ -0,0 +1,419 @@
+        """
+        if value is None:
+            return []
+        if not isinstance(value, list):
+            raise ValueError("target_ids must be a list of strings.")
+        if len(value) == 0:
+            raise ValueError("target_ids must be a non-empty list of strings.")
+        for item in value:
+            if not isinstance(item, str):
+                raise ValueError("Each target_id must be a string.")
+        return value
+
+    def add_selection(self, selection: Selection) -> None:
+        """
+        Adds a minimal selection to the list.
+
+        Args:
+            selection (Selection): The minimal selection to add.
+        """
+        self.selections.append(selection)
+
+    # override schema generation to ensure it's the way we want it
+    @classmethod
+    def model_json_schema(cls, **kwargs):
+        schema = super().model_json_schema(**kwargs)
+
+        schema["title"] = "Decision Point Value Selection List"
+        schema["$schema"] = "https://json-schema.org/draft/2020-12/schema"
+        schema["$id"] = (
+            "https://certcc.github.io/SSVC/data/schema/v2/Decision_Point_Value_Selection-2-0-0.schema.json"
+        )
+        schema["description"] = (
+            "This schema defines the structure for representing selected values from SSVC Decision Points. "
+            "Each selection list can have multiple selection objects, each representing a decision point, and each "
+            "selection object can have multiple selected values when full certainty (i.e., a singular value selection) "
+            "is not available."
+        )
+
+        non_required_fields = [
+            "name",
+            "description",
+            "target_ids",
+            "resources",
+            "references",
+        ]
+
+        # remove non-required fields from the required list
+        if "required" in schema and isinstance(schema["required"], list):
+            schema["required"] = [
+                field
+                for field in schema["required"]
+                if field not in non_required_fields
+            ]
+
+        # dive in to find all the required lists in the schema
+        # don't forget the defs
+        if "$defs" in schema:
+            for prop in schema["$defs"].values():
+                if isinstance(prop, dict) and "required" in prop:
+                    # remove non-required fields from the required list
+                    prop["required"] = [
+                        r for r in prop["required"] if r not in non_required_fields
+                    ]
+
+        # preferred order of fields, just setting for convention
+        preferred_order = [
+            "$schema",
+            "$id",
+            "title",
+            "description",
+            "schemaVersion",
+            "type",
+            "properties",
+            "required",
+            "additionalProperties",
+            "$defs",
```

Comment:
Suggesting a different order...
```suggestion
            "$schema",
            "$id",
            "title",
            "description",
            "schemaVersion",
            "type",
            "$defs",
            "required",
            "properties",
            "additionalProperties",
```

Reply:
In [CSAF](https://github.com/oasis-tcs/csaf/issues/110), we use the following order:
```
    _$comment (to be removed before publication)_
    title
    description
    type / $ref
    default
    enum
    format
    pattern
    maxLength
    minLength
    maxItems
    minItems
    maxProperties
    minProperties
    required
    uniqueItems
    properties
    items
    additionalProperties
    unevaluatedProperties
    examples
 ```